src/dawn/SIR/ASTExpr.h

```diff
@@ -115,7 +115,7 @@ class UnaryOperator : public Expr {
 
   void setOperand(const std::shared_ptr<Expr>& operand) { operand_ = operand; }
   const std::shared_ptr<Expr>& getOperand() const { return operand_; }
-  const char* getOp() const { return op_; }
+  const char* getOp() const { return op_.c_str(); }
```

Comment:
do we still need cons char* as a retval or would the switch to strings be useful also here?

Reply:
Given that is is a const getter there is not much benefit from returning a `const std::string&`.